File: wk04/verification/day_to_year_test.py
from day_to_year import day_to_year

def test_day_to_year():
    assert day_to_year(1) == 1970 # "January 1st 1970"
    assert day_to_year(366) == 1971 # "January 1st 1971"
    assert day_to_year(365 + 365 + 1) == 1972 # "January 1st 1972"
    assert day_to_year(365 + 365 + 366 + 1) == 1973 # "January 1st 1973"

File: wk08/b.coverage/day-to-year.test.ts
import { dayToYear } from './day-to-year';

describe('day to year tests', () => {
  test.each([
    { days: 1, year: 1970 }, // "January 1st 1970"
    { days: 365 + 1, year: 1971 }, // "January 1st 1971"
    { days: 365 + 365 + 1, year: 1972 }, // "January 1st 1972"
    { days: 365 + 365 + 366 + 1, year: 1973 }, // "January 1st 1973"

    // Catch bug
    { days: 365 + 365 + 366, year: 1972 }, // "December 31st 1972"

  ])('dayToYear($days) => $year', ({ days, year }) => {
    expect(dayToYear(days)).toEqual(year);
  });
});
